### PR TITLE
#2144 Document Builder. Button element.

### LINKED ISSUE #1976: Document Builder Prototype
Build the first version of the Document Builder Prototype.
https://balsamiq.cloud/s7ffzg5/pu105es/r4C1C

The Preview panel
1. shows the preview
2. allows adding a child element to a container
3. should clearly indicate that an element is a container and can have children
4. should clearly indicate the current (selected) element

The Config panel allows the adjustment of settings of the current element.

The goal of this task is to get a rough prototype so we can learn/see what the considerations are.

Document renderer issue #1872

### LINKED ISSUE #2144: Document Builder. Button element.
Add support for the Button element with effect pipeline.

### PR DESCRIPTION
**Pull request by BALEHOK**
Closes #2144 
Closes #1976 

Editor and Preview
image

Side panel (button rendered in a List element)
image

### CODE REVIEW

**Comment by twschiller** [comment]
@BALEHOK build failing due to a type error in one spot

File: src/components/documentBuilder/createNewElement.test.ts [hunk 1]
```diff
@@ -96,8 +97,23 @@ test("sets default config for block", () => {
...
 test("sets default config for button", () => {
+  const confettiEffect = new ConfettiEffect();
+  const expectedConfig = {
+    title: "Confetti",
+    onClick: {
+      __type__: "pipeline",
+      __value__: [
+        {
+          id: confettiEffect.id,
+          instanceId: expect.any(String),
```
Comment (twschiller) [inline src/components/documentBuilder/createNewElement.test.ts:108]: Cool, didn't realize you could embed expect into objects

I think we'll need to think through how instanceIds work on embedded pipelines. Our save code in the Page Editor has logic to strip out instance ids

But I suspect they aren't getting stripped out from !pipeline expressions on save

Comment (BALEHOK) [reply src/components/documentBuilder/createNewElement.test.ts:108]: True, they aren't, can be seen in the Workshop. We probably don't need them at the creation step at all.

Comment (twschiller) [reply src/components/documentBuilder/createNewElement.test.ts:108]: I would exclude them on creation for now and then we can figure out the best way to re-add them for logging/error reporting as part of the larger "embedded pipeline" discussion

File: src/components/documentBuilder/documentBuilderTypes.ts [hunk 2]
```diff
@@ -71,8 +72,27 @@ export function isPipelineElement(
   return element.type === "pipeline";
 }
 
+export type ButtonDocumentConfig = {
+  title: string | Expression;
+  variant?: string | Expression;
+  // Default size type coming from Bootstrap Button
+  size?: "sm" | "lg" | Expression<"sm" | "lg">;
```
Comment (twschiller) [inline src/components/documentBuilder/documentBuilderTypes.ts:79]: I'm surprised there isn't an md too?

**Review by twschiller — APPROVED** [review]
This is working for me, go ahead an merge when ready

We'll have to figure out how to improve the layout in the config section. It's hard to grok the layout of the button config

The  "Action button clicked" message in the preview is maybe a bit misleading since the reason to click it is to select the button?

**Pull request merged**